File: src/Algorithms/Lists/Rotation_Point/rotation-point.test.ts
// @ts-nocheck

import { TestSolutions } from "@Utils/test-solutions";

import { solutions } from "./rotation-point";

describe("Rotation Point", () => {
  TestSolutions(solutions, (solution) => {
    expect(solution(["cape", "cake"])).toBe(1);
    expect(solution(["grape", "orange", "plum", "radish", "apple"])).toBe(4);
    expect(
      solution([
        "ptolemaic",
        "retrograde",
        "supplant",
        "undulate",
        "xenoepist",
        "asymptote",
        "babka",
        "banoffee",
        "engender",
        "karpatka",
        "othellolagkage",
      ])
    ).toBe(5);
    expect(
      solution([
        "broccoli",
        "cabbage",
        "cauliflower",
        "leek",
        "radish",
        "asparagus",
        "beetroot",
        "cucumber",
        "eggplant",
      ])
    ).toBe(5);

    expect(
      solution([
        "broccoli",
        "cabbage",
        "cauliflower",
        "artichoke",
        "asparagus",
        "beetroot",
        "cucumber",
        "eggplant",
        "leek",
        "radish",
      ])
    ).toBe(3);
  });
});

File: src/Algorithms/Lists/Rotation_Point/rotation_point_test.py
from rotation_point import solution


def run_assertions(solution):
    assert solution(["cape", "cake"]) == 1
    assert solution(["grape", "orange", "plum", "radish", "apple"]) == 4
    assert solution(["ptolemaic", "retrograde", "supplant", "undulate", "xenoepist",
                     "asymptote", "babka", "banoffee", "engender", "karpatka", "othellolagkage"]) == 5
    assert solution(["broccoli", "cabbage", "cauliflower", "leek",
                     "radish", "asparagus", "beetroot", "cucumber", "eggplant"]) == 5
    assert solution(["broccoli", "cabbage", "cauliflower", "artichoke",
                     "asparagus", "beetroot", "cucumber", "eggplant", "leek", "radish"]) == 3


def test_rotation_point():
    run_assertions(solution)
